### PR TITLE
Implement readonly WebSocket client for HMD

### LINKED ISSUE #205: HMD - Websocket Client
**Is your feature request related to a problem? Please describe.**

The HMD needs to have a WebSocket client in order to receive updates from the MCP. The updates will have data identical to the JSON response body from the TSS. This WebSocket client will append data to the HMD's own datastore.

**Describe the solution you'd like**

An WebSocket client which can connect to the `ws://mcp:8181/ws/events` endpoint.

**Describe alternatives you've considered**

No other alternatives considered as the MCP facilitates communication via WebSocket connection.

**Additional context**

Will need to support WebRTC connection in the future for HoloLens and Rover video feeds.

### PR DESCRIPTION
**Pull request by dignojrteogalbo**
# Description

Updated .gitignore to ignore Unity files. Added `WebSocketClient.cs`, `EventManager.cs`, `BaseArsisEvent.cs`, and `Telemetry.cs` classes.

Fixes #205

## Bug Fix or Feature

- [ ] Bug
- [X] Feature
- [ ] Other

# Testing Instructions

From the project directory (e.g. `ARSIS/`)

- [ ] `cd ARSIS-Telemetry-GroundControl`
- [ ] `docker-compose up --build`
- [ ] Start the Unity project
- [ ] Press play on the Unity editor
- [ ] Debug console should print out received events
- [ ] Debug console prints out `Telemetry event` for telemetry events
- [ ] Debug console prints out `ARSIS.EventManager.BaseArsisEvent` for all other unimplemented events

## Linting

- [ ] I've linted my code

### CODE REVIEW

**Comment by cburnout** [comment]
Can we do this in 2 different PRs. one to update the git ignore and the tracked files and one for just the websocket?

**Comment by dignojrteogalbo** [comment]
Yeah I can open another one to commit the updated .gitignore and then merge this branch to master once the other once gets in.

File: ARSIS-Unity/Assets/Scripts/EventManager/WebSocketClient.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        /// <summary>
+        /// Performs reflection on the JSON payload and returns an object with superclass BaseArsisEvent.
+        /// </summary>
+        /// <param name="json"></param>
+        /// <returns>BaseArsisEvent or null if failed to parse JSON.</returns>
+        private BaseArsisEvent PerformReflection(string json)
+        {
+            BaseArsisEvent unknownEvent = JsonUtility.FromJson<BaseArsisEvent>(json);
+            if (unknownEvent == null) return null;
+            if (unknownEvent.type.Equals(Telemetry.Type))
```
Comment (cburnout) [inline ARSIS-Unity/Assets/Scripts/EventManager/WebSocketClient.cs:70]: I think here we can create a dictionary or do something that does this not by checking the type but by using generics to unjson it.

**Review by cburnout — CHANGES_REQUESTED** [review]

File: ARSIS-Unity/Assets/Scripts/EventManager/EventTypes/BaseArsisEvent.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    /// <summary>
+    /// This is an interface that is used to ensure a level of
+    /// uniformity with what events can do.
+    /// </summary>
+    [System.Serializable]
+    public class BaseArsisEvent
+    {
+        public static readonly Dictionary<string, Type> EventTypes = new() {
+            { "telemetry", typeof(Telemetry) },
+        };
```
Comment (Aetheris743) [inline ARSIS-Unity/Assets/Scripts/EventManager/EventTypes/BaseArsisEvent.cs:18]: What would you think about putting somthing like this in the `EventManager` to autogenerate the types?
```csharp
...
// stores reflection generated types
private Dictionary<string, Type> eventDictionary = new Dictionary<string, Type>();
```
```csharp
private struct TmpFormat
{
    public string type;
    public string data;
}

private static string EncodeEvent(BaseArsisEvent eventToEncode)
{
    string name = eventToEncode.GetType().Name;
    string body = JsonUtility.ToJson(eventToEncode);
    return JsonUtility.ToJson(new TmpFormat { type = name, data = body });
}

private BaseArsisEvent DecodeEvent(string json)
{
    TmpFormat tmp = JsonUtility.FromJson<TmpFormat>(json);
    Type type = eventDictionary[tmp.type];
    return (BaseArsisEvent)JsonUtility.FromJson(tmp.data, type);
}

```
```csharp

void Start() {
    ...
    // initalize the dict
    Type target = typeof(BaseArsisEvent);
    Assembly assembly = Assembly.GetAssembly(target);

    var types = assembly.GetTypes().Where(t => t.IsSubclassOf(target));

    foreach (var type in types)
    {
        eventDictionary.Add(type.Name, type);
    }
    ...
}

```

Comment (Aetheris743) [reply ARSIS-Unity/Assets/Scripts/EventManager/EventTypes/BaseArsisEvent.cs:18]: nvm, I think I missed your `DoReflection` method

**Review by cburnout — APPROVED** [review]

**Pull request merged**